### PR TITLE
[#3526] Add support for MSI to JS samples and generators - QnA samples

### PR DESCRIPTION
**Pull request by MartinLuccanera**
Addresses # 3526

## Description
This PR updates the QnA sample bots adding support for MSI and SingleTenant App types.

### Detailed Changes
- Updated the **_botbuilder_** dependency to the RC0 version.
- Updated **_index.js_** to pass the new parameters AppType and AppTenantId to the 
- Added the new settings _MicrosoftAppType_ and _MicrosoftAppTenantId_ to the **_env_** file.

This PR updates the following samples:
- JavaScript
   - 11.qnamaker
   - 49.qnamaker-all-features.

## Testing
The following images show one of the bots working with User Assigned MSI configuration.
image

### CODE REVIEW

File: samples/javascript_nodejs/11.qnamaker/.env [hunk 1]
```diff
@@ -1,5 +1,9 @@
+MicrosoftAppType=
 MicrosoftAppId=
 MicrosoftAppPassword=
+MicrosoftAppTenantId=
+
 QnAKnowledgebaseId=
 QnAEndpointKey=
 QnAEndpointHostName=
+DefaultAnswer=
```
Comment (ceciliaavila) [inline samples/javascript_nodejs/11.qnamaker/.env]: Why did you add this line?

Comment (sw-joelmut) [inline samples/javascript_nodejs/11.qnamaker/.env]: The DefaultAnswer is not used in this project

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: samples/javascript_nodejs/11.qnamaker/index.js [hunk 2]
```diff
@@ -25,8 +25,10 @@ const {
 const { QnABot } = require('./bots/QnABot');
 
 const credentialsFactory = new ConfigurationServiceClientCredentialFactory({
+    MicrosoftAppType: process.env.MicrosoftAppType,
```
Comment (sw-joelmut) [inline samples/javascript_nodejs/11.qnamaker/index.js]: Move the MicrosoftAppType below the MicrosoftAppPassword like we did for the rest of the samples

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Comment by ceciliaavila** [comment]
Promoted to MS in PR# 3557

**Pull request closed without merging**